HW1/P2/P2.txt

```diff
@@ -1 +1,20 @@
-# Your discussion here
+
+
+In both P2a and P2b, I use 100 partitions. In P2a the partitioning is the default for both the x and y values. 
+The image is formed from the cartesian product the the x and y value arrays. This results in a very 
+unbalanced load distribution as seen in P2a_hist.png. You can also see the results in default_events.png.
+
+In P2b, I use a different random hashing for the x and y value arrays. This results in the image worker computations
+being uniformly equal in expectation. This is because each pixel is equally likely to be in each partition unlike
+the default case which forms partitions from the 10 200x200 pixel squares (meaning the computation heavy pixels in the
+middle are likely to be in the same partitions). The worker computation is more balanced as seen in P2b_hist.png. The 
```

Comment:
Well thought out and interesting function. However, just calling partitionBy(10) on each of the axes (with a dummy key-value setup so the function works) actually produces more evenly distributed results. Just food for thought.